components/usage/pkg/db/project.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package db
+
+import (
+	"database/sql"
+	"github.com/google/uuid"
+	"gorm.io/datatypes"
+	"time"
+)
+
+type Project struct {
+	ID       uuid.UUID      `gorm:"primary_key;column:id;type:char;size:36;" json:"id"`
+	Name     string         `gorm:"column:name;type:varchar;size:255;" json:"name"`
+	Config   datatypes.JSON `gorm:"column:config;type:text;size:65535;" json:"config"`
+	CloneURL string         `gorm:"column:cloneUrl;type:varchar;size:255;" json:"cloneUrl"`
+	Slug     sql.NullString `gorm:"column:slug;type:varchar;size:255;" json:"slug"`
+	Settings sql.NullString `gorm:"column:settings;type:text;size:65535;" json:"settings"`
+
+	AppInstallationID string `gorm:"column:appInstallationId;type:varchar;size:255;" json:"appInstallationId"`
+
+	CreationTime VarcharTime `gorm:"column:creationTime;type:varchar;size:255;" json:"creationTime"`
+	LastModified time.Time   `gorm:"column:_lastModified;type:timestamp;default:CURRENT_TIMESTAMP(6);" json:"_lastModified"`
```

Comment:
**Question**: In TypeScript, we don't surface `_lastModified` in `DBProject`[[1](https://github.com/gitpod-io/gitpod/blob/44313757a7abad373916269bc72f0e2935613275/components/gitpod-db/src/typeorm/entity/db-project.ts)] -- so it's an "internal" DB field that we never interact with directly. Do we still want to handle it here?

Reply:
We can always use `_` as the name to make it defined, but not settable or usable. I personally don't see any need to omit it at this stage. IN practice, it's not settable anyway as it's set by MySQL but it can be useful for reading and checking when it was in fact last changed.